data-structures/ts/linked-lists-1.ts

```diff
@@ -0,0 +1,21 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import LinkedList from './lib/linked-list';
+
+export function getHead<T>(list: LinkedList<T>): T | null {
+  return list.data;
+}
+
+export function updateHead<T>(list: LinkedList<T>, value: T | null): void {
+  list.data = value;
+}
+
+export function withoutHead<T>(list: LinkedList<T>): LinkedList<T> | null {
+  if (!list.data === null) return null;
```

Comment:
What's the purpose of this? There are several problems here:
1. `list.data` is not related to the linked list structure. It doesn't matter what is in it. What matters is `list.next`
2. Given that, it may be surprising that this even works. It does, though, because the `if` statement is written incorrectly: The `!` binds tighter than `===` so it will be applied first, meaning that `!list.data` will be the first thing evaluated. It will evaluate to a boolean. It will be `true` if `list.data` is falsy, and `false` if `list.data` is truthy. Then that boolean will be compared to `null`. Since a boolean and a null can never be equal, the expression will always be false. Therefore, this entire statement is doing nothing. What you meant to write is `if (list.data !== null) return null;`. If you had done that, your tests would have failed and hopefully you would have realized that you don't want that check here at all.